.github/workflows/pre-commit-check.yaml

```
@@ -0,0 +1,27 @@
+# If this workflow failed in your PR, run `pre-commit install` and commit your files again.
+# See instructions in the README.md in the repo root for more details.
+name: Run pre-commit to check that formatting is correct
+on:
+  push:
+    branches: [master, main]
+  pull_request:
+
+jobs:
+  pre_commit:
+    name: "pre-commit"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@1af3b93b6815bc44a9784bd300feb67ff0d1eeb3 # ratchet:actions/checkout@v6.0.0
+
+      - uses: actions/setup-python@e797f83bcb11b83ae66e0230d6156d7c80228e7c # ratchet:actions/setup-python@v6.0.0
+        with:
+          python-version: "3.14"
+          cache: "pip"
+
+      # Only run it on changed files with: https://github.com/pre-commit/action/issues/7
+      - id: changed-files
+        uses: tj-actions/changed-files@24d32ffd492484c1d75e0c0b894501ddb9d30d62 # ratchet:tj-actions/changed-files@v47
+
+      - uses: pre-commit/action@2c7b3805fd2a0fd8c1884dcaebf91fc102a13ecd # ratchet:pre-commit/action@v3.0.1
+        with:
+          extra_args: pip-compile --files ${{ steps.changed-files.outputs.all_changed_files }}
```

I'm pretty sure that `pip-compile` is the name of a specific pre-commit hook ID to run (rather than running all as is default). It doesn't exist in this repo, so I think it'll crash. We want to run all hooks, so we can just remove it:

```suggestion
          extra_args: --files ${{ steps.changed-files.outputs.all_changed_files }}
```